lib/ansible/modules/cloud/vmware/vmware_guest_powerstate.py

```diff
@@ -64,6 +64,12 @@
     - Date and time in string format at which specificed task needs to be performed.
     - "The required format for date and time - 'dd/mm/yyyy hh:mm'."
     - Scheduling task requires vCenter server. A standalone ESXi server does not support this option.
+  wait_for_state_change:
```

Comment:
If this is not a boolean, the name should be more clear (`wait_for_state_change` sounds like a true/false value).

Suggest to make it `state_change_timeout`, description can stay